src/gun_http.erl

```diff
@@ -632,6 +634,7 @@ send_request(State=#http_state{socket=Socket, transport=Transport, version=Versi
 		headers => Headers
 	},
 	EvHandlerState1 = EvHandler:request_start(RequestEvent, EvHandlerState0),
+	%% @todo Handle send errors.
```

Comment:
This sounds like the only remaining Transport:send that isn't handled. Let's handle it? Shouldn't be too difficult as this function is only called 3 times. We could do `SendResult = Transport:send` and then only change what is returned. Maybe return `{SendResult, Authority,.... EvHandlerState}` and handle it in headers/request/ws_upgrade functions. In the error condition we need to keep the CookieStore and EvHandlerState but can ignore the rest.

Reply:
Done.